### DISCUSSION TITLE
"Online payment service", Code = 68 ein weiterer Kandidat für TInvoicePaymentMeansCode?

### DISCUSSION THREAD

**Discussion by dsshad** [LandrixSoftware/XRechnung-for-Delphi#14] [Ideas]
Hallo,
bei TInvoicePaymentMeansCode steht als Kommentar

> //TODO prüfen, ob es noch mehr gibt + Link zu den Codes

Wäre "Online payment service", Code = 68 nicht ein weiterer oft genutzter Kandidat dafür, um PayPal, Sofortüberweisung / Klarna o.ä. abzudecken?

**Comment by landrix** [comment]
Wäre zu prüfen, es werden nicht immer alle möglichen Werte der Liste in XRechnung unterstützt. Wäre dann auch wichtig zu wissen, wo denn die Kontodaten / PayPal-Adresse etc hinterlegt werden. Da habe ich keine Erfahrung. Hast Du ein Beispiel?

**Reply by dsshad** [reply]
Nein, habe leider kein Beispiel. Hatte "nur" aus Nutzersicht nachgefragt, sprich wissend, dass das häufig genutzt wird, aber nicht unterstützt wird. Dachte wenn es die Codes gibt, dann werden diese auch vom XRechnung standard unterstützt, aber da lag ich wohl falsch.

Habe jetzt versucht, etwas dazu zu finden, aber erfolglos. Habe nur eine Seite gefunden, wo beide Begriffe im passenden Zusammenhang vorkommen, aber die ist weniger hilfreich, denke ich.

Werde es mit einer direkten Anfrage bei der KosIT probieren - scheint ja vom aktuellen Standard nicht unterstützt zu werden.

Werde berichten, wenn ich etwas erfahre.

**Comment by landrix** [comment]
https://github.com/LandrixSoftware/ZUGFeRD-for-Delphi/blob/main/intf.ZUGFeRDPaymentMeansTypeCodes.pas

evtl. wäre Typ 3 noch etwas in der Richtung?

**Reply by dsshad** [reply]
An Typ 3 dachte ich auch zuerst, aber dann schien mir von der Beschreibung her 68 passender zu sein.
Hatte nach "automated clearing house" (darum geht es bei 3 laut der originalen Beschreibung) gesucht und die Ergebnisse klangen nicht nach Dienstleistern wie PayPal & Co, denke ich:
https://en.wikipedia.org/wiki/Automated_clearing_house
https://wirtschaftslexikon.gabler.de/definition/automated-clearing-house-ach-30813

**Comment by landrix** [comment]
Laut 
https://projekte.kosit.org/xrechnung/xrechnung/-/issues/76
scheint 68 nicht dabei zu sein

**Comment by dsshad** [comment]
Habe eine Antwort erhalten:

> zur Befüllung von BT-81 „Payment Means Type Code“ wird auf die Codeliste UNTDID 4461 verwiesen, deren Verwendung durch die Prüfregel BR-CL-16 geprüft wird. Hierbei handelt es sich um eine technische Regel, welche Bestandteil der von der CEN herausgegebenen Schematron Komponente (den Validierungsartefakten für die EN 16931, auf der der Standard XRechnung aufbaut) ist. Diese werden durch XRechnung implementiert. Einschränkungen bezüglich der zulässigen Codes aus dieser Liste werden in XRechnung nicht vorgenommen. Zur Verwendung bestimmter Codes sind gesonderte Bedingungen definiert (u.a. die deutschen Geschäftsregeln BR-DE-23, BR-DE-24 und BR-DE-25). Bezüglich des Codes 68 ist dies jedoch nicht der Fall. 
> 
> Kurz gesagt: der Code 68 ist in Payment means type code (BT-81) zulässig. In unserer Testsuite stehen mit dem Testfall 01.19a im Ordner test/business-cases/standard entsprechende Beispieldateien zur Verfügung. Die aktuellste Version der Testsuite steht unter [Release Testsuite release in version 2024-06-20 compatible with XRechnung 3.0.x · itplr-kosit/xrechnung-testsuite · GitHub](https://github.com/itplr-kosit/xrechnung-testsuite/releases/tag/release-2024-06-20) oder im Bundle unter [Versionen und Bundles der XRechnung - XStandards Einkauf (xeinkauf.de)](https://xeinkauf.de/xrechnung/versionen-und-bundles/) zum Download zur Verfügung.
>

**Comment by landrix** [comment]
das ging fix, ok ich schau dass ich alles bis nächste Woche gelöst bekomme

**Comment by landrix** [comment]
erledigt
